src/test/java/org/ieeervce/api/siterearnouveau/controller/ArticlesControllerTest.java

```diff
@@ -1,109 +1,176 @@
-                                .andExpect(jsonPath("$.response.title", Matchers.equalTo(title)));
-        }
+    private static final String KEYWORD_SET = "this is a large keyword set";
+    @Mock
+    ArticleService articleService;
+
+    @Spy
+    ModelMapper modelMapper = new ModelMapper();
+
+    @Spy
+    ObjectMapper objectMapper = new ObjectMapper().findAndRegisterModules();
+
+    @InjectMocks
+    ArticlesController articlesController;
+
+    MockMvc mvc;
+
+    @BeforeEach
+    void setup() {
+        mvc = MockMvcBuilders.standaloneSetup(articlesController).build();
+        objectMapper.findAndRegisterModules();
+    }
+
+    @Test
+    void testList() throws Exception {
+
+        Integer articleId = 1;
+        Integer eventCategory = 45;
+        String title = "Article " + 1;
+        String author = "Author " + 1;
+        String content = "Content " + 1;
+
+        var article = new Article();
+        article.setArticleId(articleId);
+        article.setTitle(title);
+        article.setAuthor(author);
+        article.setAddedDateTime(LocalDateTime.now());
+        article.setEventCategory(eventCategory);
+        article.setKeywords(KEYWORD_SET);
+        article.setContent(content);
+
+        when(articleService.list()).thenReturn(Collections.singletonList(article));
+
+        mvc.perform(MockMvcRequestBuilders.get("/api/article"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.ok", Matchers.equalTo(true)))
+                .andExpect(jsonPath("$.response", Matchers.iterableWithSize(1)))
+
+                .andExpect(jsonPath("$.response[0].articleId", Matchers.equalTo(articleId)))
+                .andExpect(jsonPath("$.response[0].eventCategory", Matchers.equalTo(eventCategory)))
+                .andExpect(jsonPath("$.response[0].author", Matchers.equalTo(author)))
+                .andExpect(jsonPath("$.response[0].addedDateTime", Matchers.notNullValue()))
+                .andExpect(jsonPath("$.response[0].content", Matchers.equalTo(content)))
+                .andExpect(jsonPath("$.response[0].keywords", Matchers.equalTo(KEYWORD_SET)))
+                .andExpect(jsonPath("$.response[0].title", Matchers.equalTo(title)));
+    }
+
+    @Test
+    void testGetArticleById() throws Exception {
+
+        Integer articleId = 1;
+        Integer eventCategory = 45;
+        String title = "Article " + 1;
+        String author = "Author " + 1;
+        String content = "Content " + 1;
+
+        Article article = new Article();
+        article.setArticleId(articleId);
+        article.setTitle(title);
+        article.setAuthor(author);
+        article.setAddedDateTime(LocalDateTime.now());
+        article.setEventCategory(eventCategory);
+        article.setKeywords(KEYWORD_SET);
+        article.setContent(content);
+        when(articleService.getArticleById(1)).thenReturn(article);
+
+        mvc.perform(MockMvcRequestBuilders.get("/api/article/1"))
```

Comment:
The id can be passed as string concatenation

Reply:
You're right. 
On further thought, uri variables would be better. Updated.